CRM/Core/Controller.php

```diff
@@ -457,8 +457,8 @@ public function addPages(&$stateMachine, $action = CRM_Core_Action::NONE) {
       if ($options) {
         $$stateName->setOptions($options);
       }
-      if (property_exists($$stateName, 'urlPath') && isset($_GET[CRM_Core_Config::singleton()->userFrameworkURLVar])) {
-        $$stateName->urlPath = explode('/', $_GET[CRM_Core_Config::singleton()->userFrameworkURLVar]);
+      if (property_exists($$stateName, 'urlPath')) {
```

Comment:
This is kinda confusing so talking through it....

Basically this line is checking the CRM_Core_StateMachine has a urlPath AND the url contains (for drupal) q=blah and the url check is being removed